apps/agentstack-sdk-ts/src/client/a2a/protocol/schemas.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2025 © BeeAI a Series of LF Projects, LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import z from 'zod';
+
+import { MessageRole } from './types';
+
+export const textPartSchema = z.object({
+  kind: z.literal('text'),
+  text: z.string(),
+  metadata: z.record(z.string(), z.unknown()).nullish(),
+});
+
+export const fileWithBytesSchema = z.object({
+  bytes: z.string(),
+  mimeType: z.string().nullish(),
```

Comment:
I believe all the `nullish` here is wrong.

The protocol defines the values as `undefined` not `nullish`.

This won't work in environments where plain a2a is used. 

See https://a2a-protocol.org/latest/specification/#57-field-presence-and-optionality for further reference. cc @jezekra1